plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestCorruptedIcebergTable.java

```diff
@@ -56,63 +65,41 @@ public void testCorruptedTableLocation()
             throws Exception
     {
         String tableName = "test_corrupted_table_location_" + randomNameSuffix();
+        SchemaTableName schemaTableName = SchemaTableName.schemaTableName(getSession().getSchema().orElseThrow(), tableName);
         assertUpdate("CREATE TABLE " + tableName + " (id INT, country VARCHAR, independence ROW(month VARCHAR, year INT))");
         assertUpdate("INSERT INTO " + tableName + " VALUES (1, 'INDIA', ROW ('Aug', 1947)), (2, 'POLAND', ROW ('Nov', 1918)), (3, 'USA', ROW ('Jul', 1776))", 3);
 
-        Path tableLocation = Path.of(getTableLocation(tableName));
+        Path tableLocation = Path.of(getTableLocation(getQueryRunner(), tableName));
         // break the table by deleting all its files including metadata files
         deleteRecursively(tableLocation, ALLOW_INSECURE);
 
         // Assert queries fail cleanly
-        assertThatThrownBy(() -> query("TABLE " + tableName))
-                .hasMessageContaining("Failed to open input stream for file:");
```

Comment:
If you end-up using `assertQueryFails` (as in the second commit), use it also in the first commit.

Reply:
When I use `assertQueryFails` then `QueryAssertions` does not handle the exception whose cause is not `TrinoException` that is why I ended up using `assertThatThrownBy`. Please let me know if `QueryAssertions` is supposed to handle exceptions whose cause is not `TrinoException`.

Exception which I am getting when query fails is `io.trino.testing.QueryFailedException`.

```java
assertQueryFails("TABLE " + tableName, "Failed to open input stream for file:.*");
```

```
java.lang.IllegalArgumentException: Exception does not have TrinoException cause

	at io.trino.testing.QueryAssertions.lambda$getTrinoExceptionCause$4(QueryAssertions.java:513)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.trino.testing.QueryAssertions.getTrinoExceptionCause(QueryAssertions.java:513)
	at io.trino.testing.QueryAssertions.lambda$assertQueryFails$3(QueryAssertions.java:443)
	at org.assertj.core.api.ThrowingConsumer.accept(ThrowingConsumer.java:30)
	at org.assertj.core.api.AbstractAssert.catchOptionalAssertionError(AbstractAssert.java:882)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.assertj.core.api.AbstractAssert.satisfiesForProxy(AbstractAssert.java:873)
	at org.assertj.core.api.AbstractAssert.satisfies(AbstractAssert.java:862)
	at io.trino.testing.QueryAssertions.assertQueryFails(QueryAssertions.java:443)
	at io.trino.testing.AbstractTestQueryFramework.assertQueryFails(AbstractTestQueryFramework.java:436)
	at io.trino.plugin.iceberg.TestCorruptedIcebergTable.testCorruptedTableLocation(TestCorruptedIcebergTable.java:67)
```

Reply:
Added `assertQueryFailsIncorrectly` in `TestCorruptedIcebergTable.java`